src/highlighters.cc

```diff
@@ -519,16 +520,26 @@ UniquePtr<Highlighter> create_line_highlighter(HighlighterParameters params, Hig
 }
 
 const HighlighterDesc column_desc = {
-    "Parameters: <value string> <face>\n"
-    "Highlight the column given by evaluating <value string> with <face>",
-    {}
+    "Parameters: [-ruler <character>] <column> <face>\n"
+    "Highlight the column <column> with <face>",
+    {
+        { { "ruler", { ArgCompleter{}, "replace empty or whitespace cells with the given character. When provided, <face> is not applied to non-empty cells" } } },
+        ParameterDesc::Flags::SwitchesOnlyAtStart, 2, 2
+    },
 };
 UniquePtr<Highlighter> create_column_highlighter(HighlighterParameters params, Highlighter*)
 {
-    if (params.size() != 2)
-        throw runtime_error("wrong parameter count");
+    ParametersParser parser{params, column_desc.params};
 
-    auto func = [col_expr=params[0], facespec=parse_face(params[1])]
+    auto positionals = parser.positionals_from(0);
+    auto ruler = parser.get_switch("ruler");
+    if (ruler.value_or(" ").char_length() > 1)
+        throw runtime_error("-ruler expects a single character");
+
+    auto ruler_provided = static_cast<bool>(ruler);
```

Comment:
I'd prefer `ruler.has_value()` here, and I think we could rename "ruler_provided" with what this bool is used for: highlight_non_blank, so essentially to `bool highlight_non_blank = not ruler.has_value()`

Reply:
Hmm, `has_value` doesn't seem defined for `Optional`, only for `HashMap`.

Reply:
`std::optional::has_value()` is C++ canon so it would be fine to add,
The reason why we don't have it yet is: when we want to call `explicit operator bool`, we basically always use C-style casts (`(bool)someexpr`) instead of `static_cast`. Maybe keep things consistent and make a sweeping change later if we really want that

Reply:
@krobelus so your suggestion here is to just do `(bool)ruler`, right?